File: 101.对称二叉树/isSymmetric.go
package problem0101

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isSymmetric(root *TreeNode) bool {
	return isMirrorNode(root, root)
}

func isMirrorNode(left *TreeNode, right *TreeNode) bool {
	if left == nil && right == nil {
		return true
	}
	if left == nil || right == nil {
		return false
	}
	return left.Val == right.Val && isMirrorNode(left.Left, right.Right) && isMirrorNode(left.Right, right.Left)
}

File: 101.对称二叉树/isSymmetric.js
function isSymmetric(root) {
	const isMirror = (left, right) => {
		if (left === null && right === null) return true
		if (left === null || right === null) return false
		return left.val === right.val && isMirror(left.left, right.right) && isMirror(left.right, right.left)
	}
	return isMirror(root.left, root.right)
}